### PR TITLE
Fixed error sending more than 256 fragments <master> [7404]

### PR DESCRIPTION
**Pull request by richiware**
The async thread was not awaken to send the next unsent fragments after sending the first 256. This PR fixes this.

Refactored as a port of #973

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1255/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1042/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1252/)

File: include/fastdds/rtps/common/CDRMessage_t.h [hunk 1]
```diff
@@ -196,6 +205,8 @@ struct RTPS_DllAPI CDRMessage_t final
     uint32_t pos;
     //!Max size of the message.
     uint32_t max_size;
+    //!Size allocated on buffer. May be higher than max_size.
+    uint32_t reserved_size;
```
Comment (IkerLuengo) [inline include/fastdds/rtps/common/CDRMessage_t.h]: I can only see `max_size` and `reserved_size` differing after `reserve()`, and in that case it seems that `max_size` can end up being larger than `reserverd_size`, not the contrary

Even when the `realloc` fails, we set the `max_size` to the attempted size leaving the `reserved_size` with the old (lower) value. 

What is the intention keeping a separate `max_size`?

Comment (MiguelCompany) [reply include/fastdds/rtps/common/CDRMessage_t.h]: This is about calling `reserve()` first with a big size and then with a shorter size. In that case we dont't want the reallocation, but we want the shorter size being assigned to `max_size`

Comment (IkerLuengo) [reply include/fastdds/rtps/common/CDRMessage_t.h]: It makes sense

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1489/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1260/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1487/)

**Pull request merged**